File: src/main/java/com/aesp/global/common/exception/GeneralException.java
package com.aesp.global.common.exception;

import com.aesp.global.common.code.BaseErrorCode;
import com.aesp.global.common.response.ResponseDto;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class GeneralException extends RuntimeException {

	private final BaseErrorCode code;

	public ResponseDto.ErrorReasonDto getErrorReasonDto() {
		return this.code.getReason();
	}

	public ResponseDto.ErrorReasonDto getErrorReasonHttpStatus() {
		return this.code.getReasonHttpStatus();
	}
}

File: core/core-common/src/main/kotlin/io/say/better/core/common/exception/GeneralException.kt
package io.say.better.core.common.exception

import io.say.better.core.common.code.BaseErrorCode
import io.say.better.core.common.response.ResponseDto.ErrorReasonDto

open class GeneralException(
    private val code: BaseErrorCode,
) : RuntimeException() {
    val errorReasonDto: ErrorReasonDto
        get() = this.code.reason

    val errorReasonHttpStatus: ErrorReasonDto
        get() = this.code.reasonHttpStatus
}
